roles/app/templates/nginx/sites-available/readthedocs-main.conf

```diff
@@ -37,74 +37,21 @@ server {
         add_header X-Deity docs;
     }
 
-    location ^~ /api/ {
-        proxy_pass http://{{ rtd_app_server }}:8002;
-        proxy_buffering off;
-        proxy_set_header Host $host;
-        proxy_set_header X-Real-IP $remote_addr;
-        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-        proxy_set_header X-Forwarded-Host $host;
-        access_log {{ nginx_logs }}/readthedocs-api.log host;
-    }
-
-    location ^~ /build/ {
-        proxy_pass http://{{ rtd_app_server }}:8002;
-        proxy_buffering off;
-        proxy_set_header Host $host;
-        proxy_set_header X-Real-IP $remote_addr;
-        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-        proxy_set_header X-Forwarded-Host $host;
-        access_log {{ nginx_logs }}/readthedocs-api.log host;
-    }
-
-    location ^~ /bitbucket {
-        proxy_pass http://{{ rtd_app_server }}:8002;
-        proxy_buffering off;
-        proxy_set_header Host $host;
-        proxy_set_header X-Real-IP $remote_addr;
-        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-        proxy_set_header X-Forwarded-Host $host;
-        access_log {{ nginx_logs }}/readthedocs-api.log host;
-    }
-
-    location ^~ /github {
+    location ~* /(api|build|bitbucket|github) {
         proxy_pass http://{{ rtd_app_server }}:8002;
-        proxy_buffering off;
-        proxy_set_header Host $host;
-        proxy_set_header X-Real-IP $remote_addr;
-        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-        proxy_set_header X-Forwarded-Host $host;
+        include /etc/nginx/proxy_defaults;
         access_log {{ nginx_logs }}/readthedocs-api.log host;
     }
 
     # Path handled by django
     location = / {
-        add_header X-Redirct-From Nginx;
-        add_header X-Deity docs;
-        try_files $uri $uri/ @fallback;
-        error_page 403 404 500 = @fallback;
-    }
-
-    location ^~ /admin {
-        add_header X-Redirct-From Nginx;
-        add_header X-Deity docs;
-        try_files $uri $uri/ @fallback;
-        error_page 403 404 500 = @fallback;
+        include /etc/nginx/fallback_defaults;
     }
 
-    location ^~ /dashboard {
-        add_header X-Redirct-From Nginx;
-        add_header X-Deity docs;
-        try_files $uri $uri/ @fallback;
-        error_page 403 404 500 = @fallback;
-    }
-
-    location ^~ /projects {
-        try_files $uri $uri/ @fallback;
-        error_page 403 404 500 = @fallback;
-        add_header X-Subdomain-TryFiles True;
```

Comment:
this was configured differently from the other fallback, see X-Served vs X-Redirct-From and X-Subdomain-TryFiles. It looked wrong though.

Reply:
The header was wrong